app/lib/brexit_checker/results_audiences.rb

```diff
@@ -12,17 +12,54 @@ def populate_business_groups(audience_actions, selected_criteria)
     def populate_citizen_groups(audience_actions, selected_criteria)
       return [] if audience_actions.blank? || selected_criteria.blank?
 
-      BrexitChecker::Group.load_all.each_with_object([]) do |group, result|
-        selected_actions = group.actions & audience_actions
-        unless selected_actions.empty?
-          result << {
+      audience_actions.each_with_object([]) do |action, grouped_actions|
+        next if action.grouping_criteria.empty?
+
+        group_keys =
+          multiple_grouping_criteria?(action) ? selected_group(action, selected_criteria) : action.grouping_criteria
+        group_keys.each do |key|
+          group = all_groups.select { |g| g.key == key }.first
+          next if grouped_actions.select { |actions| actions[:group] == group }.any?
```

Comment:
you can collapse this to be `next if grouped_actions.any? { |actions| actions[:group] == group }` I think, and it should have the same advantages as `detect` above :-)

Reply:
thank you!